src/redux/books/booksSlice.js

```diff
@@ -3,19 +3,50 @@
 import { createSlice } from '@reduxjs/toolkit';
 
 const initialState = {
-  books: [],
+  booksGallery: // Initial state:
+  [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
 };
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
     addBook: (state, actions) => {
-      state.books = [...state.books, actions.payload];
+      const itemId = 'item'.concat((state.booksGallery.length + 1).toString());
+      const book = {
+        ...actions.payload,
+        item_id: itemId,
+      };
+
+      return {
+        booksGallery: [...state.booksGallery, book],
+      };
```

Comment:
- [ ] Please why are you creating a new book here? :blush: Remember that an incoming `action payload` is an object that already has the `itemId` so there is no need for you to create an itemId and use that `itemId` to create a new book :smiling_face_with_tear: I think the only thing you have to do here is to update the state by appending the incoming action payload like this :point_right: `state.books = [...state.books, action.payload]`. I suggest that if you have any logic to do by setting an id for an item then it should not be here.

- [ ] I suggest you should use `action` instead of `actions` since the function is accepting only one action at a time :smile: